packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2039,6 +2037,12 @@ export class ContainerRuntime
 		// There might be no change of state due to Container calling this API after loading runtime.
 		const changeOfState = this._connected !== connected;
 		const reconnection = changeOfState && !connected;
+
+		// We need to flush the ops currently collected by Outbox to the PendingStateManager to preserve original order upon calling "replayPendingStates"
```

Comment:
Should we do this closer to the `replayPendingStates` below? I mean in the same conditional?

Reply:
Oh, I think not, as by the time that happens, connected is true, so the outbox will actually send them. Can you add a comment to be more explicit here and call out the fact that this code relies on the shouldSend parameter provided to the outbox (connected = false) and the idea is to roundtrip all batches to the pendingstatemanager via the outbox without actually sending it?